src/log.js

```diff
@@ -747,23 +747,55 @@ class MemLogger extends FormattedLoggerBase {
  *
  * @class
  * @param {Object} opts – Optional, named parameters
- * @param {string} [opts.level='silly'] The minimum log level to sent to loggly
+ * @param {string} [opts.level='silly'] The minimum log level to sent to the logger
+ * @param {Logger} [opts.logger = rootLogger] The helix logger to use
  * @param {Function} [opts.filter=identity] Will be given every log message to perform
  *   arbitrary transformations; must return either another valid message object or undefined
  *   (in which case the message will be dropped).
+ * @param {object} [opts.fields] Additional log fields to add to every log message.
  */
 class InterfaceBase {
+  /**
+   * Helper method for setting defaults recursively.
+   * todo: add to ferrum?
+   * @private
+   */
+  static _deepsetdefault(src, dst) {
+    Object.entries(src).forEach(([key, value]) => {
+      if (key in dst) {
+        if (typeof value === 'object') {
+          InterfaceBase._deepsetdefault(value, dst[key]);
+        }
+      } else {
+        dst[key] = value;
+      }
+    });
+  }
+
   constructor({
-    logger = rootLogger, level = 'silly', filter = identity, ...unknown
+    logger = rootLogger, level = 'silly', filter = identity, fields = {}, ...unknown
   } = {}) {
-    assign(this, { logger, level, filter });
+    assign(this, {
+      logger, level, filter, fields,
```

Comment:
Please use a more specific name; `defaultFields` instead of fields might be suitable…

Please introduce the fields property to all Loggers, not just the interfaces.

And finally please update the documentation of the Interface & Logger traits with the property;

(that they `SHOULD` (as in RFC should) support the property; if they do support the property they MUST perform a shallow merge/setdefault into the message AFTER applying the filters – This is to retain the ability for filters to set default parameters)

Reply:
I rather see this a functionality on the interface, not on the loggers.

Reply:
Loggers have the same basic interface as far as this feature is concerned; please, let's keep things consistent.